src/base/PolymeshTransactionBase.ts

```diff
@@ -451,6 +453,34 @@ export abstract class PolymeshTransactionBase<
     };
   }
 
+  /**
+   * Get the latest processed block from the database
+   *
+   * @note uses the middleware
+   */
+  private async getLatestBlockFromDb(): Promise<BigNumber> {
```

Comment:
(nit) its getting it from a GraphQL server, not a DB. "FromDb" makes it seem like there is a direct DB connection to me.

```suggestion
  private async getLatestBlockFromMiddleware(): Promise<BigNumber> {
```
or
```suggestion
  private async getLatestMiddlewareBlock(): Promise<BigNumber> {
```